src/antidote.erl

```diff
@@ -148,37 +147,68 @@ update_objects(Updates, TxId) ->
 update_objects(Clock, Properties, Updates) ->
     update_objects(Clock, Properties, Updates, false).
 
+-spec update_objects(snapshot_time() | ignore , term(), [{bound_object(), op_name(), op_param()}], boolean()) ->
+    {ok, snapshot_time()} | {error, reason()}.
+update_objects(_Clock, _Properties, [], _StayAlive) ->
+    {ok, vectorclock:new()};
 update_objects(Clock, _Properties, Updates, StayAlive) ->
-    Operations = lists:map(
-                   fun({{Key, Type, Bucket}, Op, OpParam}) ->
-                           {update, {{Key, Bucket}, Type, {Op,OpParam}}}
-                   end,
-                   Updates),
-    SingleKey = case Operations of
-                    [_O] -> %% Single key update
-                        case Clock of
-                            ignore -> true;
-                            _ -> false
-                        end;
-                    [_H|_T] -> false
-                end,
-    case SingleKey of
-        true ->  %% if single key, execute the fast path
-            [{update, {K, T, Op}}] = Operations,
-            case append(K, T, Op) of
-                {ok, {_TxId, [], CT}} ->
-                    {ok, CT};
+    case check_and_format_ops(Updates) of
```

Comment:
This changes make the interactive coordinator execute an update_objects transaction in a static way, similarly to how the static coordinator was doing it, but unified the code in the interactive one, which includes the parallel updates optimization.